src/PostgREST/Auth.hs

```diff
@@ -39,16 +39,16 @@ data JWTAttempt = JWTInvalid JWTError
                 deriving (Eq, Show)
 
 {-|
-  Receives the JWT secret (from config) and a JWT and returns a map
+  Receives the JWT secret and audience (from config) and a JWT and returns a map
   of JWT claims.
 -}
-jwtClaims :: Maybe JWK -> BL.ByteString -> IO JWTAttempt
-jwtClaims _ "" = return $ JWTClaims M.empty
-jwtClaims secret payload =
+jwtClaims :: Maybe JWK -> Text -> BL.ByteString  -> IO JWTAttempt
+jwtClaims _ "" "" = return $ JWTClaims M.empty
+jwtClaims secret audience payload =
   case secret of
     Nothing -> return JWTMissingSecret
     Just jwk -> do
-      let validation = defaultJWTValidationSettings
+      let validation = set audiencePredicate (==  audience) defaultJWTValidationSettings
```

Comment:
As you pointed out, the error says we're trying to compare StringOrURI with Text, so we want to find a way to convert `audience` so we can compare it.

The first thing to note in debugging this issue is that our cabal file specifies
[jose < 0.6](https://github.com/begriffs/postgrest/blob/master/postgrest.cabal#L69), so we need to be sure when looking at jose code on hackage we look at v0.5.0.5.

My next step was scanning the docs on hackage for a function to convert the types. This one turns up:

https://hackage.haskell.org/package/jose-0.5.0.5/docs/Crypto-JWT.html#v:fromString

Making this change fixes the error you encountered:

```haskell
let validation = set audiencePredicate (==  fromString audience) defaultJWTValidationSettings
```

Reply:
Thank you very much for the patient help! This was exactly what I needed, and I had been looking at the wrong version of jose docs and getting very confused.